core/src/main/scala/org/apache/spark/HttpServer.scala

```diff
@@ -155,6 +156,7 @@ private[spark] class HttpServer(
       throw new ServerStateException("Server is already stopped")
     } else {
       server.stop()
+      Option(server.getThreadPool).collect { case x: LifeCycle => x }.foreach(_.stop())
```

Comment:
I think I'll use the if `isInstanceOf` as it's the most obvious on the intention and doesn't need an empty catch all case (which is a bit overkill in here since there is only one condition to match).

Just for the sake of discussion of style, I think it depends on whether you interpret in the imperative way "if some condition then do something" vs the functional way "create option -> filter by condition -> apply operation", which can be a forever debate :) I myself lean against the functional way for doing simple task (like there one liner in here), as it is more concise; but the imperative way when need to implement more complicated logical flow.